examples/vision/semisupervised_simclr.py

```diff
@@ -0,0 +1,658 @@
+
+        # only the probe metrics are logged at test time
+        return {m.name: m.result() for m in self.metrics[2:]}
+
+
+# the contrastive model is pretrained for half of the epochs
+pretraining_model = ContrastiveModel()
+pretraining_model.compile(
+    contrastive_optimizer=keras.optimizers.Adam(),
+    probe_optimizer=keras.optimizers.Adam(),
+)
+
+pretraining_history = pretraining_model.fit(
+    train_dataset, epochs=num_epochs // 2, validation_data=test_dataset
+)
+print(
+    "Maximal validation accuracy: {:.2f}%".format(
+        max(pretraining_history.history["val_p_acc"]) * 100
+    )
+)
+
+"""
+## Supervised finetuning of the pretrained encoder
+
+We then finetune the encoder on the labeled examples for 30 epochs, by attaching
+a single randomly initalized fully connected classification layer on its top.
+"""
+
+# the contrastive model is then finetuned for the other half of the epochs
+finetuning_model = keras.Sequential(
+    [
+        layers.Input(shape=(image_size, image_size, image_channels)),
+        get_augmenter(**classification_augmentation),
+        pretraining_model.encoder,
+        layers.Dense(10),
+    ],
+    name="finetuning_model",
+)
+finetuning_model.compile(
+    optimizer=keras.optimizers.Adam(),
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    metrics=[keras.metrics.SparseCategoricalAccuracy(name="acc")],
+)
+
+finetuning_history = finetuning_model.fit(
+    labeled_train_dataset, epochs=num_epochs // 2, validation_data=test_dataset
+)
+print(
+    "Maximal validation accuracy: {:.2f}%".format(
+        max(finetuning_history.history["val_acc"]) * 100
+    )
+)
+
+"""
+## Comparison against the baseline
+"""
+
+
+# the classification accuracies of the baseline and the pretraining + finetuning process:
+def plot_training_curves(pretraining_history, finetuning_history, baseline_history):
+    pretraining_epochs = len(pretraining_history.history["val_p_acc"])
+    finetuning_epochs = len(finetuning_history.history["val_acc"])
+
+    for metric_key, metric_name in zip(["acc", "loss"], ["accuracy", "loss"]):
+        plt.figure(figsize=(8, 5), dpi=100)
+        plt.plot(
+            baseline_history.history[f"val_{metric_key}"], label="supervised baseline"
+        )
+        plt.plot(
+            pretraining_history.history[f"val_p_{metric_key}"],
+            label="self-supervised pretraining",
+        )
+        plt.plot(
+            list(range(pretraining_epochs, pretraining_epochs + finetuning_epochs)),
+            finetuning_history.history[f"val_{metric_key}"],
+            label="supervised finetuning",
+        )
+        plt.legend()
+        plt.title(f"Classification {metric_name} during training")
+        plt.xlabel("epochs")
+        plt.ylabel(f"validation {metric_name}")
+
+
+plot_training_curves(pretraining_history, finetuning_history, baseline_history)
+
+"""
+By comparing the training curves, we can see that by using contrastive
```

Comment:
Suggestion:

- Avoid "By.... by..." with "By.... when..." as follows:

```suggestion
By comparing the training curves, we can see that when using contrastive
```